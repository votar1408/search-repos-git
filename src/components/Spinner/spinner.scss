.root {
    display: flex;
    position: absolute;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.2);
    top: 0;
    left: 0;
}

.root__spinner {
    width: '100px';
}

.spinner-img {
    max-width: '100%';
}
