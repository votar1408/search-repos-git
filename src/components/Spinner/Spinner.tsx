import './spinner.scss';

import React, {FC} from 'react';

import spinner from '../../assets/svg/spinner.svg';

export const Spinner: FC = () => {
    return (
        <div className="root">
            <div className="root__spinner">
                <img className="spinner-img" src={spinner} alt="spinner" />
            </div>
        </div>
    );
};
