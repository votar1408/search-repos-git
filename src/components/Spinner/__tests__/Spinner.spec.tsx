import React from 'react';

import {
    shallow,
    ShallowWrapper
} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {Spinner} from '../Spinner';

describe('Spinner', () => {
    const wrapper: ShallowWrapper = shallow(<Spinner />);

    it('should render correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        expect(wrapper).toBeTruthy();
    });

    it('should contains <img/>', () => {
        expect(wrapper.find('img').length).toEqual(1);
    });

    it('should img contains src', () => {
        expect(wrapper.find('img').prop('src')).toEqual('spinner.svg');
    });
});
