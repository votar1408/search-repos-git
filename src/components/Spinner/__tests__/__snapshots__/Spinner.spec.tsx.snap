// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render correctly 1`] = `
<div
  className="root"
>
  <div
    className="root__spinner"
  >
    <img
      alt="spinner"
      className="spinner-img"
      src="spinner.svg"
    />
  </div>
</div>
`;
